src/rust/engine/fs/store/src/immutable_inputs.rs

```diff
@@ -49,7 +55,74 @@ impl ImmutableInputs {
   }
 
   /// Returns an absolute Path to immutably consume the given Digest from.
-  async fn path(&self, directory_digest: DirectoryDigest) -> Result<PathBuf, StoreError> {
+  pub(crate) async fn path_for_file(
```

Comment:
The implementation is mostly duped. @stuhood graciously offered to take a stab at a helper

Reply:
One of these days!

Reply:
:sweat_smile: It might be once we remove the provided `immutable_input_digests` one of them goes away? Unclear.